tests/johnson_alahi_li_2016/test_modules.py

```diff
@@ -0,0 +1,291 @@
+import itertools
+
+import pytest
+
+import pytorch_testing_utils as ptu
+import torch
+from torch import nn
+
+import pystiche
+from pystiche.misc import to_2d_arg
+from pystiche_papers.johnson_alahi_li_2016 import modules
+from pystiche_papers.utils import ResidualBlock
+
+
+def is_url(url):  # TODO: implement this
+    return True
+
+
+def test_select_url(subtests):
+
+    configs = (
+        ("candy", True, True),
+        ("composition_vii", True, False),
+        ("feathers", True, True),
+        ("la_muse", True, False),
+        ("la_muse", True, True),
+        ("mosaic", True, True),
+        ("starry_night", True, False),
+        ("the_scream", True, True),
+        ("the_wave", True, False),
+        ("udnie", True, True),
+    )
+    for style, impl_params, instance_norm in configs:
+        with subtests.test(style):
+            url = modules.select_url(
+                style,
+                weights="author",
+                impl_params=impl_params,
+                instance_norm=instance_norm,
+            )
+            assert is_url(url)
+
+
+def test_select_url_no_valid_style():
+    impl_params = (True, False)
+    instance_norm = (True, False)
+    configs = itertools.product(impl_params, instance_norm)
+    for impl_params, instance_norm in configs:
+        with pytest.raises(RuntimeError):
+            modules.select_url(
+                "no_valid_style",
+                weights="pystiche",
+                impl_params=impl_params,
+                instance_norm=instance_norm,
+            )
+
+
+def test_get_conv(subtests):
+    in_channels = out_channels = 3
+    kernel_size = 3
+    stride = 1
+    padding = (None, 1, (1, 1))
+    upsample = (True, False)
+    configs = itertools.product(padding, upsample)
+    for padding, upsample in configs:
+        conv_module = modules.get_conv(
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride=stride,
+            padding=padding,
+            upsample=upsample,
+        )
+
+        with subtests.test("module"):
+            assert isinstance(
+                type(conv_module),
+                type(nn.ConvTranspose2d) if upsample else type(nn.Conv2d),
+            )
```

Comment:
We don't need a subtest for this, since this is the most basic test we can do. If this fails, the other will most likely also fail.

```suggestion
        assert isinstance(
            type(conv_module), type(nn.ConvTranspose2d) if upsample else type(nn.Conv2d),
        )
```